tesseract/tesseract_planning/tesseract_motion_planners/test/simple_planner_lvs_interpolation.cpp

```diff
@@ -0,0 +1,367 @@
+ * limitations under the License.
+ */
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <gtest/gtest.h>
+#include <boost/filesystem.hpp>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract/tesseract.h>
+#include <tesseract_motion_planners/simple/simple_motion_planner.h>
+#include <tesseract_motion_planners/simple/step_generators/lvs_interpolation.h>
+
+using namespace tesseract;
+using namespace tesseract_planning;
+
+bool DEBUG = false;
+
+std::string locateResource(const std::string& url)
+{
+  std::string mod_url = url;
+  if (url.find("package://tesseract_support") == 0)
+  {
+    mod_url.erase(0, strlen("package://tesseract_support"));
+    size_t pos = mod_url.find('/');
+    if (pos == std::string::npos)
+    {
+      return std::string();
+    }
+
+    std::string package = mod_url.substr(0, pos);
+    mod_url.erase(0, pos);
+    std::string package_path = std::string(TESSERACT_SUPPORT_DIR);
+
+    if (package_path.empty())
+    {
+      return std::string();
+    }
+
+    mod_url = package_path + mod_url;
+  }
+
+  return mod_url;
+}
+
+class TesseractPlanningSimplePlannerLVSInterpolationUnit : public ::testing::Test
+{
+protected:
+  Tesseract::Ptr tesseract_ptr_;
+  ManipulatorInfo manip_info_;
+  std::vector<std::string> joint_names_;
+
+  void SetUp() override
+  {
+    tesseract_scene_graph::ResourceLocator::Ptr locator =
+        std::make_shared<tesseract_scene_graph::SimpleResourceLocator>(locateResource);
+    Tesseract::Ptr tesseract = std::make_shared<Tesseract>();
+    boost::filesystem::path urdf_path(std::string(TESSERACT_SUPPORT_DIR) + "/urdf/lbr_iiwa_14_r820.urdf");
+    boost::filesystem::path srdf_path(std::string(TESSERACT_SUPPORT_DIR) + "/urdf/lbr_iiwa_14_r820.srdf");
+    EXPECT_TRUE(tesseract->init(urdf_path, srdf_path, locator));
+    tesseract_ptr_ = tesseract;
+
+    manip_info_.manipulator = "manipulator";
+    joint_names_ = tesseract_ptr_->getManipulatorManager()->getFwdKinematicSolver("manipulator")->getJointNames();
+  }
+};
+
+TEST_F(TesseractPlanningSimplePlannerLVSInterpolationUnit, InterpolateStateWaypoint_JointJoint_Freespace)  // NOLINT
+{
+  PlannerRequest request;
+  request.tesseract = tesseract_ptr_;
+  request.env_state = tesseract_ptr_->getEnvironment()->getCurrentState();
+  JointWaypoint wp1(joint_names_, Eigen::VectorXd::Zero(7));
+  JointWaypoint wp2(joint_names_, Eigen::VectorXd::Ones(7));
+  PlanInstruction instr(wp1, PlanInstructionType::FREESPACE, "DEFAULT", manip_info_);
+
+  auto composite = LVSInterpolateStateWaypoint(wp1, wp2, instr, request, ManipulatorInfo(), 3.14, 0.5, 1.57, 5);
+  for (const auto& c : composite)
+  {
+    EXPECT_TRUE(isMoveInstruction(c));
+    EXPECT_TRUE(isStateWaypoint(c.cast_const<MoveInstruction>()->getWaypoint()));
+  }
+  const auto* mi = composite.back().cast_const<MoveInstruction>();
+  EXPECT_TRUE(wp2.isApprox(mi->getWaypoint().cast_const<StateWaypoint>()->position, 1e-5));
+}
```

Comment:
Sorry to be nit picky, but these unit tests don't test the most important features of these functions - that the LVS logic works correctly and that the interpolation is happening correctly. I bring this up because I was having an issue when switching to use this code. When looking at the tests here, they don't really guarantee that it is working at all. Has this code been used on some project so that we are reasonably certain it is doing what it should be doing?

Reply:
@marrts Can you take a look at this?

Reply:
Specifically, interpolation seems to be working fine. However on a joint joint freespace moves I seem to always get 11 points. Again, it could be my issue. I just don't know for sure since I don't think these tests test that.

Reply:
I'll take a look at this, I think I've only really tested with cartesian to cartesian in an actual planning environment.

Reply:
See #398